csharp/LogicalColumnReader.cs

```diff
@@ -199,6 +201,22 @@ public override int ReadBatch(Span<TElement> destination)
             return _batchReader.ReadBatch(destination);
         }
 
+        public override long Skip(long numRowsToSkip)
+        {
+            var isArray = typeof(TElement).IsArray;
+            if (isArray)
+            {
+                var arrayReader = _batchReader as IArraySkippable;
```

Comment:
This feels unelegant, but I'm not sure how to make it better. Thoughts welcome.

Reply:
I think you might want to add a Skip method to `ILogicalBatchReader` instead and implement it for all the different reader types, as eg. this doesn't look like it will work correctly for an array within a nested reader? For simple cases like the direct and scalar reader they can implement this just by calling skip on the physical column reader.

Reply:
That was my first implementation but wasn't sure it was relevant for all other reader types. Will revert back to that.